fastlane/Fastfile

```diff
@@ -111,4 +120,37 @@ platform :ios do
       sh('bundle', 'exec', 'pod', 'repo', 'update')
     end
   end
+
+  desc "Post-release"
+  private_lane :post_release do |options|
+    version = options[:version].to_s
+    # increment plugin lock files by calling pod update on each
+    plugin_root = '/Users/distiller/amplify-ios/AmplifyPlugins'
+  
+    sh('bundle', 'exec', 'pod', 'repo', 'update')
```

Comment:
I'm confused by the sequence:

- We begin with the repo version strings at 1.0.0, with Podfile.lock at 1.0.0
- We bump repo versions to 1.0.1
- We release the pods at tag `v1.0.1`; but with Podfile.lock still pointing to 1.0.0
- We bump the Podfile.lock to 1.0.1

Doesn't that cause an inconsistency in the versioning, or am I misunderstanding the process? If I'm misunderstanding, a README or documentation in this Fastfile itself would help clarify for future developers

Reply:
The podfile.locks are only used when running the source code locally or running the unit tests, so they don't have to be incremented before the release. I.e. when installing the pods via `pod install`, the lock files aren't downloaded. They can only be retrieved from the git repo directly. 

We can't call `pod update` on these until the pods in question have been pushed to trunk. We could iterate over them and set the values to the latest versions manually, but then the spec checksums would be incorrect, e.g. https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/API/Podfile.lock#L92

Reply:
In the case you describe, the developer would be checking out from `main` or `release`, but it's a common task to troubleshoot errors on a particular release by checking out a tag. E.g.:

- Customer reports a bug on 1.0.1
- Developer checks out source at tag `v1.0.1`
- Developer does a `pod install`
- CocoaPods downloads pods according to the Podfile.lock checked in with tag `v1.0.1`, which point to `v1.0.0` versions
- Hilarity and/or chaos ensues

Am I still misunderstanding? Shouldn't we also move the `v1.0.1` tag from the original commit to the commit that contains the newly updated Podfile.lock?

Reply:
Got it. I wasn't aware of that development flow.
We can move the remote tag in the post_release step then 👍🏻

Reply:
Actually, thinking about this more, the issue is that we're a monorepo with dependencies to pods contained in our own repo.

The right way to fix this might be to do a `pod update` before each tag, push, and cocoapods push.